Dont remove version.c and ensure the version.c is written.

Upstream-status: Pending

1. not remove version.c, since making $(obj).depend target needs version.c;
otherwise the below error will happen:
    arm-wrs-linux-gnueabi-gcc -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a   -MQ version.o version.c >.depend.version
    arm-wrs-linux-gnueabi-gcc: error: version.c: No such file or directory
    arm-wrs-linux-gnueabi-gcc: fatal error: no input files

2. when version.c is generated, force changed blocks to disk; otherwise
version.c maybe a empty file when other process is reading it, and the below
error will happen:
    | arch/arm/lib/libarm.o: In function `display_banner':
    | git/arch/arm/lib/board.c:356: undefined reference to `get_lsi_version'
    | arm-wrs-linux-gnueabi-ld: error: required section '.rel.plt' not found in the linker script
    | arm-wrs-linux-gnueabi-ld: final link failed: Invalid operation


Signed-off-by: Roy.Li <rongqing.li@windriver.com>
---
 board/lsi/common/Makefile | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/board/lsi/common/Makefile b/board/lsi/common/Makefile
index 1a863e2..59962db 100644
--- a/board/lsi/common/Makefile
+++ b/board/lsi/common/Makefile
@@ -73,9 +73,6 @@ version.c: version
 
 $(obj)version.o: version.c
 	$(CC) $(CFLAGS) -c -o $@ $<
-ifdef CONFIG_SPL_BUILD
-	rm -f $<
-endif
 
 version: version.c.in
 	@git --version >/dev/null 2>&1 ; \
@@ -89,7 +86,8 @@ version: version.c.in
 			VERSION_STRING="UNKNOWN" ; \
 		fi ; \
 	fi ; \
-	sed -e s/GIT_VERSION_STRING/$$VERSION_STRING/ version.c.in >version.c
+	sed -e s/GIT_VERSION_STRING/$$VERSION_STRING/ version.c.in >version.c; \
+	sync
 
 # defines $(obj).depend target
 include $(SRCTREE)/rules.mk
-- 
1.9.1

